discord/utils.py

```diff
@@ -605,6 +605,8 @@ def _get_as_snowflake(data: Any, key: str) -> Optional[int]:
     except KeyError:
         return None
     else:
+        if value == '':
+            return None
         return value and int(value)
```

Comment:
There's no reason to do a separate check here, just modify the already existing one that covered None when discord actually sent what they were supposed to 🙄 
```suggestion
        return int(value) if value else None
```

Reply:
this could also alternatively be fixed by changing line 604 to 

```py
        value = data[key] or None
```
but I think this would be worse for the normal case (haven't checked)